nix/flake/modules/server.nix

```diff
@@ -0,0 +1,877 @@
+overlay: { config, lib, pkgs, ... }:
```

Comment:
Key change: module by itself takes an overlay to later reference the package without a a call to callPackage.

The rest of changes to this file (`diff ../module-server.nix modules/server.nix`) are mostly `nixpkgs-fmt`.